engine/runtime/src/main/java/org/enso/interpreter/runtime/builtin/Builtins.java

```diff
@@ -250,39 +247,98 @@ public void initializeBuiltinsIr(FreshNameSupply freshNameSupply, Passes passes)
     }
   }
 
-  @CompilerDirectives.TruffleBoundary
-  public void readAllBuiltinNodes(ModuleScope scope) {
-    Reflections reflections = new Reflections("org.enso.interpreter.node.expression.builtin");
-    Set<String> subtypes = reflections.get(Scanners.SubTypes.of(Scanners.TypesAnnotated.with(NodeInfo.class)));
-    subtypes.stream().forEach(className -> {
+  private void readBuiltinsMetadata(ModuleScope scope) {
+    ClassLoader classLoader = getClass().getClassLoader();
+    FileSystem fs = null;
+    Stream<Path> builtinMetaPath;
+    try {
+      URI resource = classLoader.getResource(MethodDefinition.META_PATH).toURI();
```

Comment:
How is the _cache_ structured? The content is clear - each line contains a pair `owner.method:nodeClass`. But why there is the `FileSystem.walk`? It'd be simpler to have a just a single resources which contains all the necessary registrations.